### PR TITLE
Add nanoflow property

### PR DESCRIPTION
**Pull request by jseremba**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
     },
     "exclude": [
-        "dist/",
-        "node_modules/"
+      "dist/",
+      "node_modules/"
     ]
-}
+  }
+
```
Comment (Andries-Smit) [inline tsconfig.json]: missing newline at the end of file

File: src/components/ProgressBarContainer.ts [hunk 2]
```diff
@@ -29,7 +30,12 @@ interface ProgressBarContainerState {
...
 
-type OnClickOptions = "doNothing" | "showPage" | "callMicroflow";
+interface Nanoflow {
+    nanoflow: object[];
+    paramsSpec: { Progress: string };
+}
+
+type OnClickOptions = "doNothing" | "showPage" | "callMicroflow" | "callNanoflow";
 
 export default class ProgressBarContainer extends Component<ProgressBarContainerProps, ProgressBarContainerState> {
```
Comment (Andries-Smit) [inline src/components/ProgressBarContainer.ts]: Please also update `ProgressBar.webmodeler.ts` file. (applies for all widgets)

File: tsconfig.json (hunk 1 above)
Comment (jseremba) [reply tsconfig.json]: Done

File: Gruntfile.js [hunk 3]
```diff
@@ -3,30 +3,26 @@ const webpack = require("webpack");
...
-webpackConfig.forEach(function(config) {
-    webpackConfigRelease.push(merge(config, {
-        devtool: false,
-        plugins: [ new webpack.optimize.UglifyJsPlugin() ]
-    }));
-});
+const webpackConfigRelease = webpackConfig.map(config => merge(config, {
+    devtool: false,
+    mode: "production",
+    optimization: {
```
Comment (McDoyen) [inline Gruntfile.js]: Can be put on one line.

File: tests/helpers/structureMatcher.ts [hunk 4]
```diff
@@ -198,4 +201,4 @@ export function toElementStructure(node: any): ElementStructure | ElementStructu
 
 function isJasmineAny(obj: any): obj is jasmine.Any {
     return obj != null && typeof obj.asymmetricMatch === "function";
-}
+}
```
Comment (McDoyen) [inline tests/helpers/structureMatcher.ts]: Add end of line

**Review by McDoyen — CHANGES_REQUESTED** [review]

File: src/components/ProgressBarContainer.ts [hunk 5]
```diff
@@ -81,7 +87,7 @@ export default class ProgressBarContainer extends Component<ProgressBarContainer
             className: this.props.class,
             colorSwitch: this.props.textColorSwitch,
             maximumValue: this.state.maximumValue,
-            onClickAction: (this.props.onClickMicroflow || this.props.onClickPage) && this.props.mxObject
+            onClickAction: (this.props.onClickMicroflow || this.props.onClickNanoflow || this.props.onClickPage) && this.props.mxObject
```
Comment (Andries-Smit) [inline src/components/ProgressBarContainer.ts]: Should compare with `this.props.onClickOption !== "doNothing"`

Comment (McDoyen) [reply src/components/ProgressBarContainer.ts]: Done

File: tests/helpers/structureMatcher.ts (hunk 4 above)
Comment (jseremba) [reply tests/helpers/structureMatcher.ts]: Done

**Pull request merged**